include/core/nexus/HY_PointHydroNexusRemote.hpp

```diff
@@ -0,0 +1,76 @@
+#ifndef HY_POINTHYDRONEXUSREMOTEDOWNSTREAM_H
+#define HY_POINTHYDRONEXUSREMOTEDOWNSTREAM_H
+
+#ifdef NGEN_MPI_ACTIVE
+
+#include <HY_PointHydroNexus.hpp>
+#include <mpi.h>
+#include <vector>
+
+#include <unordered_map>
+
+
+/** This class representa a point nexus that can have both upstream and downstream connections to catments that are
+*   in seperate MPI processes.
+*
+*   When attempting to add upstream flows from a remote catchment a MPI_Irecv call will be generated
+*   When attempting to send flows to remote downstream a MPI_Send will be generated
+*   In either case the change in local water amounts for the time step will be recorded when the MPI operation completes */
+
+class HY_PointHydroNexusRemote : public HY_PointHydroNexus
+{
+    public:
+        /** This is map of catchment ids to MPI ranks it does not have to include any catchment that is local or any catchment that will not be
+        * communicated with */
+
+        typedef std::unordered_map <long, long> catcment_location_map_t;
```

Comment:
Check typing of catchment id, would probably be better suited as a string

Reply:
this has been changed to unordered_map<long,std::string>